kernel/device.c

```diff
@@ -9,89 +9,93 @@
 #include <sys/atomic.h>
 #include <syscall_handler.h>
 
-extern struct device __device_init_start[];
-extern struct device __device_PRE_KERNEL_1_start[];
-extern struct device __device_PRE_KERNEL_2_start[];
-extern struct device __device_POST_KERNEL_start[];
-extern struct device __device_APPLICATION_start[];
-extern struct device __device_init_end[];
+extern struct init_entry __init_start[];
+extern struct init_entry __init_PRE_KERNEL_1_start[];
+extern struct init_entry __init_PRE_KERNEL_2_start[];
+extern struct init_entry __init_POST_KERNEL_start[];
+extern struct init_entry __init_APPLICATION_start[];
+extern struct init_entry __init_end[];
 
 #ifdef CONFIG_SMP
-extern struct device __device_SMP_start[];
+extern struct init_entry __init_SMP_start[];
 #endif
 
+extern struct device __device_start[];
+extern struct device __device_end[];
+
 #ifdef CONFIG_DEVICE_POWER_MANAGEMENT
 extern u32_t __device_busy_start[];
 extern u32_t __device_busy_end[];
 #define DEVICE_BUSY_SIZE (__device_busy_end - __device_busy_start)
 #endif
 
 /**
- * @brief Execute all the device initialization functions at a given level
+ * @brief Execute all the init entry initialization functions at a given level
  *
- * @details Invokes the initialization routine for each device object
- * created by the DEVICE_INIT() macro using the specified level.
- * The linker script places the device objects in memory in the order
+ * @details Invokes the initialization routine for each init entry object
+ * created by the INIT_ENTRY_DEFINE() macro using the specified level.
+ * The linker script places the init entry objects in memory in the order
  * they need to be invoked, with symbols indicating where one level leaves
  * off and the next one begins.
  *
  * @param level init level to run.
  */
-void z_sys_device_do_config_level(s32_t level)
+void z_sys_init_run_level(s32_t level)
 {
-	struct device *info;
-	static struct device *config_levels[] = {
-		__device_PRE_KERNEL_1_start,
-		__device_PRE_KERNEL_2_start,
-		__device_POST_KERNEL_start,
-		__device_APPLICATION_start,
+	static struct init_entry *levels[] = {
```

Comment:
could be const?

Reply:
Indeed. Actually it's missing in various places.